src/ESM.sol

```diff
@@ -65,6 +67,19 @@ contract ESM {
         require(z >= x);
     }
 
+    // -- admin --
+    function file(bytes32 what, uint256 data) external {
+        require(msg.sender == proxy, "ESM/not-authorized");
```

Comment:
Yeah this is a good point. Adding the full `wards` with the pause proxy as the sole ward makes sense to me.

Reply:
yeah I was trying to avoid adding all of that and thought since the `proxy` was already available as the governance address we were concerned about denying I could use it as a shortcut.  I'm not sure what you mean by "can't be empty", seems like this can be `address(0)` or any address passed in the constructor.  Pushing a commit with the rely/deny, but that requires changing the current `deny` function since it overlaps with the auth deny.